dev/src/path.ts

```diff
@@ -158,20 +162,70 @@ abstract class Path<T> {
   compareTo(other: Path<T>): number {
     const len = Math.min(this.segments.length, other.segments.length);
     for (let i = 0; i < len; i++) {
-      if (this.segments[i] < other.segments[i]) {
-        return -1;
-      }
-      if (this.segments[i] > other.segments[i]) {
-        return 1;
+      const comparison = this.compareSegments(
+        this.segments[i],
+        other.segments[i]
+      );
+      if (comparison !== 0) {
+        return comparison;
       }
     }
-    if (this.segments.length < other.segments.length) {
+    return this.compareNumbers(
+      BigInt(this.segments.length),
+      BigInt(other.segments.length)
+    );
```

Comment:
ultra-nit [feel free to ignore]: segment lengths are usually very small positive numbers.
Doing `return Math.sign(this.segments.length - other.segments.length);` is slightly better than converting to BigInt and then comparing.